frontend/lib/app.tsx

```diff
@@ -58,6 +58,12 @@ export interface AppProps {
    * be populated with the content of the modal.
    */
   modal?: JSX.Element;
+
+  /**
+   * A render prop to render the current route. This is intended primarily
+   * for testing purposes.
+   */
+  renderRoute?: (props: RouteComponentProps<any>) => JSX.Element;
```

Comment:
This is the only change I've made to actual non-testing code.  I'm not a big fan of it--another approach was refactoring `<App>` into multiple components which would probably be smarter but also take more work--but it gets the job done for now.